graph/src/util/intern.rs

```diff
@@ -0,0 +1,658 @@
+//! Interning of strings.
+//!
+//! This module provides an interned string pool `AtomPool` and a map-like
```

Comment:
nitpick: I don't think this really is a util, it's more of a data structure as state, also intern doesn't really describe what is contained in the file so I think either a re-usable crate if possible or at least a different package name/path would benefit readability